site/docs/reference/Connectors/capture-connectors/mariadb.md

```diff
@@ -0,0 +1,314 @@
+
+## Prerequisites
+To use this connector, you'll need a MariaDB database setup with the following.
+* [`binlog_format`](https://mariadb.com/kb/en/binary-log-formats/)
+  system variable set to `ROW`.
+* [Binary log expiration period](https://mariadb.com/kb/en/using-and-maintaining-the-binary-log/#purging-log-files) set to at least 30 days (2592000 seconds) if at all possible.
+  - This value may be set lower if necessary, but we [strongly discourage](#insufficient-binlog-retention) going below 7 days as this may increase the likelihood of unrecoverable failures.
+  MariaDB's default value is 0 (no expiration).
+* A watermarks table. The watermarks table is a small "scratch space"
+  to which the connector occasionally writes a small amount of data (a UUID,
+  specifically) to ensure accuracy when backfilling preexisting table contents.
+  - The default name is `"flow.watermarks"`, but this can be overridden in `config.json`.
+* A database user with appropriate permissions:
+  - `REPLICATION CLIENT` and `REPLICATION SLAVE` [privileges](https://mariadb.com/docs/skysql/ref/es10.6/privileges/).
+  - Permission to insert, update, and delete on the watermarks table.
+  - Permission to read the tables being captured.
+  - Permission to read from `information_schema` tables, if automatic discovery is used.
+* If the table(s) to be captured include columns of type `DATETIME`, the `time_zone` system variable
+  must be set to an IANA zone name or numerical offset.
+
+### Setup
+To meet these requirements, do the following:
+
+1. Create the watermarks table. This table can have any name and be in any database, so long as the capture's `config.json` file is modified accordingly.
+```sql
+CREATE DATABASE IF NOT EXISTS flow;
+CREATE TABLE IF NOT EXISTS flow.watermarks (slot INTEGER PRIMARY KEY, watermark TEXT);
+```
+2. Create the `flow_capture` user with replication permission, the ability to read all tables, and the ability to read and write the watermarks table.
+
+  The `SELECT` permission can be restricted to just the tables that need to be
+  captured, but automatic discovery requires `information_schema` access as well.
+```sql
+CREATE USER IF NOT EXISTS flow_capture
+  IDENTIFIED BY 'secret'
+  COMMENT 'User account for Flow MariaDB data capture';
+GRANT REPLICATION CLIENT, REPLICATION SLAVE ON *.* TO 'flow_capture';
+GRANT SELECT ON *.* TO 'flow_capture';
+GRANT INSERT, UPDATE, DELETE ON flow.watermarks TO 'flow_capture';
+```
+3. Configure the binary log to retain data for 30 days, if previously set lower.
+```sql
+SET PERSIST binlog_expire_logs_seconds = 2592000;
+```
+4. Configure the database's time zone. See [below](#setting-the-mariadb-time-zone) for more information.
+```sql
+SET PERSIST time_zone = '-05:00'
+```
+
+### Setting the MariaDB time zone
+
+MariaDB's [`time_zone` server system variable](https://mariadb.com/kb/en/server-system-variables/#system_time_zone) is set to `SYSTEM` by default.
```

Comment:
I know this is copied from the MySQL equivalent, but re-reading it now it feels like it might be worth adding a bit more explanation about why we need time zone information here, and what the consequences are if it's not present. Something like:

"Flow captures `DATETIME` columns as RFC3339 timestamps, but `DATETIME` values have no associated time zone within MariaDB, so this is inferred from the system `time_zone` variable when it is set. If `time_zone` is unset or set to `SYSTEM` it will not be possible to capture tables with `DATETIME` columns, but all other column types will continue to work. Note that `TIMESTAMP` columns do not require the system `time_zone` to be set because they are normalized from the client time zone to UTC at insert time."